### ISSUE TITLE
Pretrained Models and Training Code for Table 2 Experiments

### ISSUE THREAD

**Issue by mmaaz60** [facebookresearch/Detic#32]
Hi,

Is there any plan to release the pretrained models and training code for the Detic model in Table-2 (Open-vocabulary LVIS compared to ViLD)?

Thank you

**Comment by mmaaz60** [comment]
Hi @xingyizhou,

Any update on this? Thank you

**Comment by xingyizhou** [comment]
Hi,

  Sorry for my very delayed reply. We don't plan to officially support the ViLD baseline given its super long training time. I am happy to share the model and config links here for reproducibility purposes. Please let me know if this is super urgent (e.g., for an ECCV submission). It takes time for me to dig out the model and verify them ...

Best,
Xingyi

**Comment by xingyizhou** [comment]
Hi,

  I have uploaded our ViLD configs for our [Box-supervised baseline](https://github.com/facebookresearch/Detic/blob/main/configs/BoxSup_ViLD_200e.py) and [Detic](https://github.com/facebookresearch/Detic/blob/main/configs/Detic_ViLD_200e.py). 

  Our Detic model is [here](https://drive.google.com/file/d/178T7vEbDRW5B82kZiQIPxOrsz8Mqb-B2/view?usp=sharing). The model is finetuned on the Box-supervised model as described in our paper. Unfortunately, I don't have the exact BoxSup model we used now. You can train the BoxSup model yourself using the config above or fine-tune from [this](https://drive.google.com/file/d/1RJ4rUlcVOA0BWaamq2YieDFLaNHEpMoB/view?usp=sharing) similar model (class-specific mask head, which has ~0 novel class mask mAP. It should not affect the performance of Detic fine-tuning). 

Best,
Xingyi

**Comment by mmaaz60** [comment]
Thank you @xingyizhou

**Comment by mmaaz60** [comment]
Hi @xingyizhou,

I hope you are doing good. I have tried using the `BoxSup_ViLD_200e.py` config but wasn't able to start the training. I am getting the following error,
![image](https://user-images.githubusercontent.com/55104795/157133094-3c7edb83-47c2-4134-a61f-163356c2824e.png)

I tried replacing line 38 from `conv_norm=lambda c: NaiveSyncBatchNorm(c, stats_mode="N")` to `conv_norm="SyncBN"` and then got the following error,
![image](https://user-images.githubusercontent.com/55104795/157133297-0aecfb77-3a30-4858-a7e2-01b8b5733113.png)

I will appreciate any help in solving this issue. Further, I can understand that you might be really busy these days, so please reply at your convienience. 

Thank you

**Comment by xingyizhou** [comment]
Hi,

  What is your pytorch/ cuda/ detectron2 version? What is your command for running the training? Can you run the evaluation script successfully? `conv_norm="SyncBN"` seems infeasible here as lazy config use the exact instance instead of a string.

Best,
Xingyi

**Comment by mmaaz60** [comment]
> Hi,
> 
> What is your pytorch/ cuda/ detectron2 version? What is your command for running the training? Can you run the evaluation script successfully? `conv_norm="SyncBN"` seems infeasible here as lazy config use the exact instance instead of a string.
> 
> Best, Xingyi

Hi,

Thank you for your reply. I was getting the above errors because I was using `train_net.py` for lazy configs as I was not much familiar with this type of config files. ​Using `lazy_train_net.py` solved the above errors.

I can reproduce the Detic results using the provided pretrained weights as can be seen below,
![image](https://user-images.githubusercontent.com/55104795/157559547-11ef4bd9-d6c7-4ba9-a414-09a3f583f43a.png)

I am now trying to train the Box Supervised model using the config [BoxSup_ViLD_200e.py](https://github.com/facebookresearch/Detic/blob/main/configs/BoxSup_ViLD_200e.py) and found that the training is diverging. Currently, I am using a batch size of 128, LR of 0.0004, and training on 16 V100 GPUs (e.g. I just set the num_nodes=2 in the config). What do you think could be the problem? Is it because of the batch size (I'm using 128 instead of 256), but here I decreased the LR accordingly? I can think of trying 256 batch size but that might not be feasible with the resources I have. I will appreciate your comments. Thank you

![image](https://user-images.githubusercontent.com/55104795/157559741-d5063684-f6a6-4a52-a6f0-74f917c10c74.png)

**Comment by xingyizhou** [comment]
Hi,

  I also met the diverged training, increasing the batchsize generally can help. Here are additional things I tried to stabilize training. However the provided config should be the exact one I used in our paper:

  - [Initialize classification bias](https://github.com/facebookresearch/Detic/blob/main/configs/BoxSup_ViLD_200e.py#L55) so that the default scores after sigmoid is 0.01. Some losses need it, but I didn't use it in the final version.
  - Longer warmup.
  - Make sure you used sync BN instead of BN.
  - [Turn off FP16](https://github.com/facebookresearch/detectron2/blob/main/configs/new_baselines/mask_rcnn_R_50_FPN_100ep_LSJ.py#L14).

  Please also refer to the original [ViLD repo](https://github.com/tensorflow/tpu/tree/master/models/official/detection/projects/vild) for the ViLD baseline. For this repo, I highly recommend our baseline which is MUCH resource-friendly (~18 hours one machine) and performs better.

Best,
Xingyi
